roles/lib_utils/library/oo_azure_rm_publish_image.py

```diff
@@ -194,19 +228,28 @@ def run_ansible(params):
 
         if params['state'] == 'offer':
             results = apc.manage_offer(params)
+        elif params['state'] == 'publish':
+            results = apc.publish(offer=params['offer'], emails=params['emails'])
         elif params['state'] == 'cancel_op':
-            results = apc.cancel_operation(offer=params['offer'])
+            results = apc.cancel_and_wait_for_operation(params)
         elif params['state'] == 'go_live':
             results = apc.go_live(offer=params['offer'])
         else:
-            raise AzurePublisherException('Unsupported query type: {}'.format(params['query']))
+            raise AzurePublisherException('Unsupported query type: {}'.format(params['state']))
 
         changed = False
 
-        if results.status_code in [200, 201]:
+        if results.status_code in [200, 201, 202]:
```

Comment:
change still necessary?

Reply:
Good question.  If we return the cancel_operation then it was a `202`.  We are now returning the get_operation call which is `200`.  This doesn't hurt anything. 

What would you prefer?